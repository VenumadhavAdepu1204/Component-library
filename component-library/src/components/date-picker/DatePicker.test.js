import React from 'react'
import { mount, configure, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import DatePicker from './index'

configure({ adapter: new Adapter() })

describe('DatePicker', () => {
  const app = render(
    <DatePicker
      label='date-picker'
      id='test'
      disablePreviousDates={new Date(2018, 10, 1)}
      disableAfterDates={new Date(2018, 10, 11)}
    />
  )

  test('renders properly', () => {
    expect(app).toMatchSnapshot()
  })
})

describe('Date picker methods test', () => {
  it('show calendar on click on calendar icon', () => {
    const app = mount(
      <DatePicker
        label='date-picker'
        id='test-2'
        disablePreviousDates={new Date(2018, 10, 1)}
        disableAfterDates={new Date(2018, 10, 11)}
        placeholder='DD/MM/YYYY'
      />)
    app.find('[data-test-id="datepicker-icon"]').first().simulate('click')
    const calendar = app.exists('[data-test-id="date-picker-calendar"]')
    expect(calendar).toEqual(true)
    app.unmount()
  })

  it('Click on left arrow and test the getNextMonth, getNextYear and getNextDecade methdos', () => {
    const appWrapper = mount(
      <DatePicker
        label='date-picker'
        id='test-2'
        placeholder='DD/MM/YYYY'
      />)
    appWrapper.find('[data-test-id="datepicker-icon"]').first().simulate('click')
    const calendar = appWrapper.exists('[data-test-id="date-picker-calendar"]')
    expect(calendar).toEqual(true)
    appWrapper.find('[data-test-id="right-arrow"]').first().simulate('click')
    appWrapper.find('[data-test-id="navigation-label"]').first().simulate('click')
    appWrapper.find('[data-test-id="right-arrow"]').first().simulate('click')
    appWrapper.find('[data-test-id="navigation-label"]').first().simulate('click')
    appWrapper.find('[data-test-id="right-arrow"]').first().simulate('click')
  })

  it('Click on left arrow and test the getPreviousMonth, getPerviousYear and getPreviousDecade methdos', () => {
    const testWrapper = mount(
      <DatePicker
        label='date-picker'
        id='test-2'
        placeholder='DD/MM/YYYY'
      />)
    testWrapper.find('[data-test-id="datepicker-icon"]').first().simulate('click')
    const calendar = testWrapper.exists('[data-test-id="date-picker-calendar"]')
    expect(calendar).toEqual(true)
    testWrapper.find('[data-test-id="left-arrow"]').first().simulate('click')
    testWrapper.find('[data-test-id="navigation-label"]').first().simulate('click')
    testWrapper.find('[data-test-id="left-arrow"]').first().simulate('click')
    testWrapper.find('[data-test-id="navigation-label"]').first().simulate('click')
    testWrapper.find('[data-test-id="left-arrow"]').first().simulate('click')
  })

  it('click on Year and navigation label', () => {
    const testApp = mount(
      <DatePicker
        label='date-picker'
        id='test-2'
        placeholder='DD/MM/YYYY'
      />)
    testApp.find('[data-test-id="datepicker-icon"]').first().simulate('click')
    const calendar = testApp.exists('[data-test-id="date-picker-calendar"]')
    expect(calendar).toEqual(true)
    testApp.find('[data-test-id="navigation-label"]').first().simulate('click')
    testApp.find('[data-test-id="navigation-label"]').first().simulate('click')
    testApp.find('[data-test-id="date-picker-year"]').first().simulate('click')
    testApp.unmount()
  })

  it('click on month and navigation label', () => {
    const mountApp = mount(
      <DatePicker
        label='date-picker'
        id='test-2'
        placeholder='DD/MM/YYYY'
      />)
    mountApp.find('[data-test-id="datepicker-icon"]').first().simulate('click')
    const calendar = mountApp.exists('[data-test-id="date-picker-calendar"]')
    expect(calendar).toEqual(true)
    mountApp.find('[data-test-id="navigation-label"]').first().simulate('click')
    mountApp.find('[data-test-id="date-picker-month"]').first().simulate('click')
    mountApp.unmount()
  })

  // it('click on Day and navigation label', () => {
  //   const wrapper = mount(
  //     <DatePicker
  //       label='date-picker'
  //       id='test-2'
  //       placeholder='DD/MM/YYYY'
  //     />)
  //   wrapper.find('[data-test-id="datepicker-icon"]').first().simulate('click')
  //   const calendar = wrapper.exists('[data-test-id="date-picker-calendar"]')
  //   expect(calendar).toEqual(true)
  //   wrapper.find('[data-test-id="day-remaining"]').first().simulate('click')
  //   wrapper.unmount()
  // })
})
