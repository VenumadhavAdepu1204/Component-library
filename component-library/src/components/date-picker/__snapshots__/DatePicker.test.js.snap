// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders properly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa jaOyYS",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-bwzfXH gOKpXh",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-hSdWYo bcEtTB",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "sc-eNQAEJ cHNlSk",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "sc-hMqMXs dLhBbu",
                      "data-test-id": "test-input-text-label",
                      "for": "test",
                    },
                    "children": Array [
                      Object {
                        "data": "date-picker",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "data-required": "false",
                        "data-test-id": "test",
                        "data-testid": "test",
                        "id": "test",
                        "placeholder": "DD/MM/YYYY",
                        "tabindex": "0",
                        "type": "text",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "class": "sc-eHgmQL hKmccN",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "aria-expanded": "false",
                              "aria-haspopup": "true",
                              "aria-label": "datepicker icon",
                              "class": "sc-htpNat VLjUo",
                              "data-test-id": "datepicker-icon",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "data-test-id": "undefined-icon-svg",
                                  "focusable": "false",
                                  "height": "1em",
                                  "viewBox": "0 0 32 32",
                                  "width": "1em",
                                },
                                "children": Array [
                                  Object {
                                    "attribs": Object {
                                      "d": "M21.4,9.1 C20.6,9.1 20,8.6 20,7.8 L20,2.3 C20,1.6 20.6,1 21.4,1 L21.6,1 C22.4,1 23,1.6 23,2.3 L23,7.8 C23,8.5 22.4,9.1 21.6,9.1 L21.4,9.1 Z M10.4,9.1 C9.6,9.1 9,8.6 9,7.8 L9,2.3 C9,1.6 9.6,1 10.4,1 L10.6,1 C11.4,1 12,1.6 12,2.3 L12,7.8 C12,8.5 11.4,9.1 10.6,9.1 L10.4,9.1 Z M10.3,15.1 C10.7,15.1 11,15.4 11,15.8 L11,18.5 C11,18.9 10.7,19.2 10.3,19.2 L7.6,19.2 C7.2,19.2 6.9,18.9 6.9,18.5 L6.9,15.8 C6.9,15.4 7.2,15.1 7.6,15.1 L10.3,15.1 Z M17.3,15.1 C17.7,15.1 18,15.4 18,15.8 L18,18.5 C18,18.9 17.7,19.2 17.3,19.2 L14.6,19.2 C14.2,19.2 13.9,18.9 13.9,18.5 L13.9,15.8 C13.9,15.4 14.2,15.1 14.6,15.1 L17.3,15.1 Z M24.3,15.1 C24.7,15.1 25,15.4 25,15.8 L25,18.5 C25,18.9 24.7,19.2 24.3,19.2 L21.6,19.2 C21.2,19.2 20.9,18.9 20.9,18.5 L20.9,15.8 C20.9,15.4 21.2,15.1 21.6,15.1 L24.3,15.1 Z M10.3,21.1 C10.7,21.1 11,21.4 11,21.8 L11,24.5 C11,24.9 10.7,25.2 10.3,25.2 L7.6,25.2 C7.2,25.2 6.9,24.9 6.9,24.5 L6.9,21.8 C6.9,21.4 7.2,21.1 7.6,21.1 L10.3,21.1 Z M17.3,21.1 C17.7,21.1 18,21.4 18,21.8 L18,24.5 C18,24.9 17.7,25.2 17.3,25.2 L14.6,25.2 C14.2,25.2 13.9,24.9 13.9,24.5 L13.9,21.8 C13.9,21.4 14.2,21.1 14.6,21.1 L17.3,21.1 Z M24.3,21.1 C24.7,21.1 25,21.4 25,21.8 L25,24.5 C25,24.9 24.7,25.2 24.3,25.2 L21.6,25.2 C21.2,25.2 20.9,24.9 20.9,24.5 L20.9,21.8 C20.9,21.4 21.2,21.1 21.6,21.1 L24.3,21.1 Z M25.5182724,4.1 L24.6212625,4.1 L24.6212625,7.3 C24.6212625,9 23.2259136,10.5 21.4318937,10.5 C19.6378738,10.5 18.2425249,9.1 18.2425249,7.3 L18.2425249,4.1 L13.7574751,4.1 L13.7574751,7.3 C13.7574751,9 12.3621262,10.5 10.5681063,10.5 C8.77408638,10.5 7.37873754,9.1 7.37873754,7.3 L7.37873754,4.1 L6.48172757,4.1 C3.49169435,4.1 1,6.5 1,9.5 L1,25.7 C1,28.7 3.49169435,31.1 6.48172757,31.1 L25.5182724,31.1 C28.5083056,31.1 31,28.7 31,25.7 L31,9.6 C31,6.6 28.5083056,4.1 25.5182724,4.1 Z M28.0099668,26.3 C28.0099668,27.3 27.2126246,28.2 26.2159468,28.2 L5.78405316,28.2 C4.78737542,28.2 3.99003322,27.4 3.99003322,26.3 L3.99003322,14 C3.99003322,13 4.78737542,12.1 5.78405316,12.1 L26.2159468,12.1 C27.2126246,12.1 28.0099668,12.9 28.0099668,14 L28.0099668,26.3 L28.0099668,26.3 Z",
                                      "data-test-id": "undefined-icon-path",
                                      "fill": "#007DBA",
                                      "id": "undefined-data-id",
                                    },
                                    "children": Array [],
                                    "name": "path",
                                    "namespace": "http://www.w3.org/2000/svg",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "tag",
                                    "x-attribsNamespace": Object {
                                      "d": undefined,
                                      "data-test-id": undefined,
                                      "fill": undefined,
                                      "id": undefined,
                                    },
                                    "x-attribsPrefix": Object {
                                      "d": undefined,
                                      "data-test-id": undefined,
                                      "fill": undefined,
                                      "id": undefined,
                                    },
                                  },
                                ],
                                "name": "svg",
                                "namespace": "http://www.w3.org/2000/svg",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "data-test-id": undefined,
                                  "focusable": undefined,
                                  "height": undefined,
                                  "viewBox": undefined,
                                  "width": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "data-test-id": undefined,
                                  "focusable": undefined,
                                  "height": undefined,
                                  "viewBox": undefined,
                                  "width": undefined,
                                },
                              },
                            ],
                            "name": "button",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "aria-expanded": undefined,
                              "aria-haspopup": undefined,
                              "aria-label": undefined,
                              "class": undefined,
                              "data-test-id": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "aria-expanded": undefined,
                              "aria-haspopup": undefined,
                              "aria-label": undefined,
                              "class": undefined,
                              "data-test-id": undefined,
                            },
                          },
                        ],
                        "name": "span",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "data-required": undefined,
                        "data-test-id": undefined,
                        "data-testid": undefined,
                        "id": undefined,
                        "placeholder": undefined,
                        "tabindex": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "data-required": undefined,
                        "data-test-id": undefined,
                        "data-testid": undefined,
                        "id": undefined,
                        "placeholder": undefined,
                        "tabindex": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "data-test-id": undefined,
                      "for": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "data-test-id": undefined,
                      "for": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "autocomplete": "off",
                      "data-required": "false",
                      "data-test-id": "test",
                      "data-testid": "test",
                      "id": "test",
                      "placeholder": "DD/MM/YYYY",
                      "tabindex": "0",
                      "type": "text",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "class": "sc-eHgmQL hKmccN",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "aria-expanded": "false",
                            "aria-haspopup": "true",
                            "aria-label": "datepicker icon",
                            "class": "sc-htpNat VLjUo",
                            "data-test-id": "datepicker-icon",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "data-test-id": "undefined-icon-svg",
                                "focusable": "false",
                                "height": "1em",
                                "viewBox": "0 0 32 32",
                                "width": "1em",
                              },
                              "children": Array [
                                Object {
                                  "attribs": Object {
                                    "d": "M21.4,9.1 C20.6,9.1 20,8.6 20,7.8 L20,2.3 C20,1.6 20.6,1 21.4,1 L21.6,1 C22.4,1 23,1.6 23,2.3 L23,7.8 C23,8.5 22.4,9.1 21.6,9.1 L21.4,9.1 Z M10.4,9.1 C9.6,9.1 9,8.6 9,7.8 L9,2.3 C9,1.6 9.6,1 10.4,1 L10.6,1 C11.4,1 12,1.6 12,2.3 L12,7.8 C12,8.5 11.4,9.1 10.6,9.1 L10.4,9.1 Z M10.3,15.1 C10.7,15.1 11,15.4 11,15.8 L11,18.5 C11,18.9 10.7,19.2 10.3,19.2 L7.6,19.2 C7.2,19.2 6.9,18.9 6.9,18.5 L6.9,15.8 C6.9,15.4 7.2,15.1 7.6,15.1 L10.3,15.1 Z M17.3,15.1 C17.7,15.1 18,15.4 18,15.8 L18,18.5 C18,18.9 17.7,19.2 17.3,19.2 L14.6,19.2 C14.2,19.2 13.9,18.9 13.9,18.5 L13.9,15.8 C13.9,15.4 14.2,15.1 14.6,15.1 L17.3,15.1 Z M24.3,15.1 C24.7,15.1 25,15.4 25,15.8 L25,18.5 C25,18.9 24.7,19.2 24.3,19.2 L21.6,19.2 C21.2,19.2 20.9,18.9 20.9,18.5 L20.9,15.8 C20.9,15.4 21.2,15.1 21.6,15.1 L24.3,15.1 Z M10.3,21.1 C10.7,21.1 11,21.4 11,21.8 L11,24.5 C11,24.9 10.7,25.2 10.3,25.2 L7.6,25.2 C7.2,25.2 6.9,24.9 6.9,24.5 L6.9,21.8 C6.9,21.4 7.2,21.1 7.6,21.1 L10.3,21.1 Z M17.3,21.1 C17.7,21.1 18,21.4 18,21.8 L18,24.5 C18,24.9 17.7,25.2 17.3,25.2 L14.6,25.2 C14.2,25.2 13.9,24.9 13.9,24.5 L13.9,21.8 C13.9,21.4 14.2,21.1 14.6,21.1 L17.3,21.1 Z M24.3,21.1 C24.7,21.1 25,21.4 25,21.8 L25,24.5 C25,24.9 24.7,25.2 24.3,25.2 L21.6,25.2 C21.2,25.2 20.9,24.9 20.9,24.5 L20.9,21.8 C20.9,21.4 21.2,21.1 21.6,21.1 L24.3,21.1 Z M25.5182724,4.1 L24.6212625,4.1 L24.6212625,7.3 C24.6212625,9 23.2259136,10.5 21.4318937,10.5 C19.6378738,10.5 18.2425249,9.1 18.2425249,7.3 L18.2425249,4.1 L13.7574751,4.1 L13.7574751,7.3 C13.7574751,9 12.3621262,10.5 10.5681063,10.5 C8.77408638,10.5 7.37873754,9.1 7.37873754,7.3 L7.37873754,4.1 L6.48172757,4.1 C3.49169435,4.1 1,6.5 1,9.5 L1,25.7 C1,28.7 3.49169435,31.1 6.48172757,31.1 L25.5182724,31.1 C28.5083056,31.1 31,28.7 31,25.7 L31,9.6 C31,6.6 28.5083056,4.1 25.5182724,4.1 Z M28.0099668,26.3 C28.0099668,27.3 27.2126246,28.2 26.2159468,28.2 L5.78405316,28.2 C4.78737542,28.2 3.99003322,27.4 3.99003322,26.3 L3.99003322,14 C3.99003322,13 4.78737542,12.1 5.78405316,12.1 L26.2159468,12.1 C27.2126246,12.1 28.0099668,12.9 28.0099668,14 L28.0099668,26.3 L28.0099668,26.3 Z",
                                    "data-test-id": "undefined-icon-path",
                                    "fill": "#007DBA",
                                    "id": "undefined-data-id",
                                  },
                                  "children": Array [],
                                  "name": "path",
                                  "namespace": "http://www.w3.org/2000/svg",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                  "x-attribsNamespace": Object {
                                    "d": undefined,
                                    "data-test-id": undefined,
                                    "fill": undefined,
                                    "id": undefined,
                                  },
                                  "x-attribsPrefix": Object {
                                    "d": undefined,
                                    "data-test-id": undefined,
                                    "fill": undefined,
                                    "id": undefined,
                                  },
                                },
                              ],
                              "name": "svg",
                              "namespace": "http://www.w3.org/2000/svg",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "data-test-id": undefined,
                                "focusable": undefined,
                                "height": undefined,
                                "viewBox": undefined,
                                "width": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "data-test-id": undefined,
                                "focusable": undefined,
                                "height": undefined,
                                "viewBox": undefined,
                                "width": undefined,
                              },
                            },
                          ],
                          "name": "button",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "aria-expanded": undefined,
                            "aria-haspopup": undefined,
                            "aria-label": undefined,
                            "class": undefined,
                            "data-test-id": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "aria-expanded": undefined,
                            "aria-haspopup": undefined,
                            "aria-label": undefined,
                            "class": undefined,
                            "data-test-id": undefined,
                          },
                        },
                      ],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "class": "sc-hMqMXs dLhBbu",
                        "data-test-id": "test-input-text-label",
                        "for": "test",
                      },
                      "children": Array [
                        Object {
                          "data": "date-picker",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "data-test-id": undefined,
                        "for": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "data-test-id": undefined,
                        "for": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "autocomplete": undefined,
                      "data-required": undefined,
                      "data-test-id": undefined,
                      "data-testid": undefined,
                      "id": undefined,
                      "placeholder": undefined,
                      "tabindex": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "autocomplete": undefined,
                      "data-required": undefined,
                      "data-test-id": undefined,
                      "data-testid": undefined,
                      "id": undefined,
                      "placeholder": undefined,
                      "tabindex": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "class": "sc-eHgmQL hKmccN",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "aria-expanded": "false",
                          "aria-haspopup": "true",
                          "aria-label": "datepicker icon",
                          "class": "sc-htpNat VLjUo",
                          "data-test-id": "datepicker-icon",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "data-test-id": "undefined-icon-svg",
                              "focusable": "false",
                              "height": "1em",
                              "viewBox": "0 0 32 32",
                              "width": "1em",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "d": "M21.4,9.1 C20.6,9.1 20,8.6 20,7.8 L20,2.3 C20,1.6 20.6,1 21.4,1 L21.6,1 C22.4,1 23,1.6 23,2.3 L23,7.8 C23,8.5 22.4,9.1 21.6,9.1 L21.4,9.1 Z M10.4,9.1 C9.6,9.1 9,8.6 9,7.8 L9,2.3 C9,1.6 9.6,1 10.4,1 L10.6,1 C11.4,1 12,1.6 12,2.3 L12,7.8 C12,8.5 11.4,9.1 10.6,9.1 L10.4,9.1 Z M10.3,15.1 C10.7,15.1 11,15.4 11,15.8 L11,18.5 C11,18.9 10.7,19.2 10.3,19.2 L7.6,19.2 C7.2,19.2 6.9,18.9 6.9,18.5 L6.9,15.8 C6.9,15.4 7.2,15.1 7.6,15.1 L10.3,15.1 Z M17.3,15.1 C17.7,15.1 18,15.4 18,15.8 L18,18.5 C18,18.9 17.7,19.2 17.3,19.2 L14.6,19.2 C14.2,19.2 13.9,18.9 13.9,18.5 L13.9,15.8 C13.9,15.4 14.2,15.1 14.6,15.1 L17.3,15.1 Z M24.3,15.1 C24.7,15.1 25,15.4 25,15.8 L25,18.5 C25,18.9 24.7,19.2 24.3,19.2 L21.6,19.2 C21.2,19.2 20.9,18.9 20.9,18.5 L20.9,15.8 C20.9,15.4 21.2,15.1 21.6,15.1 L24.3,15.1 Z M10.3,21.1 C10.7,21.1 11,21.4 11,21.8 L11,24.5 C11,24.9 10.7,25.2 10.3,25.2 L7.6,25.2 C7.2,25.2 6.9,24.9 6.9,24.5 L6.9,21.8 C6.9,21.4 7.2,21.1 7.6,21.1 L10.3,21.1 Z M17.3,21.1 C17.7,21.1 18,21.4 18,21.8 L18,24.5 C18,24.9 17.7,25.2 17.3,25.2 L14.6,25.2 C14.2,25.2 13.9,24.9 13.9,24.5 L13.9,21.8 C13.9,21.4 14.2,21.1 14.6,21.1 L17.3,21.1 Z M24.3,21.1 C24.7,21.1 25,21.4 25,21.8 L25,24.5 C25,24.9 24.7,25.2 24.3,25.2 L21.6,25.2 C21.2,25.2 20.9,24.9 20.9,24.5 L20.9,21.8 C20.9,21.4 21.2,21.1 21.6,21.1 L24.3,21.1 Z M25.5182724,4.1 L24.6212625,4.1 L24.6212625,7.3 C24.6212625,9 23.2259136,10.5 21.4318937,10.5 C19.6378738,10.5 18.2425249,9.1 18.2425249,7.3 L18.2425249,4.1 L13.7574751,4.1 L13.7574751,7.3 C13.7574751,9 12.3621262,10.5 10.5681063,10.5 C8.77408638,10.5 7.37873754,9.1 7.37873754,7.3 L7.37873754,4.1 L6.48172757,4.1 C3.49169435,4.1 1,6.5 1,9.5 L1,25.7 C1,28.7 3.49169435,31.1 6.48172757,31.1 L25.5182724,31.1 C28.5083056,31.1 31,28.7 31,25.7 L31,9.6 C31,6.6 28.5083056,4.1 25.5182724,4.1 Z M28.0099668,26.3 C28.0099668,27.3 27.2126246,28.2 26.2159468,28.2 L5.78405316,28.2 C4.78737542,28.2 3.99003322,27.4 3.99003322,26.3 L3.99003322,14 C3.99003322,13 4.78737542,12.1 5.78405316,12.1 L26.2159468,12.1 C27.2126246,12.1 28.0099668,12.9 28.0099668,14 L28.0099668,26.3 L28.0099668,26.3 Z",
                                  "data-test-id": "undefined-icon-path",
                                  "fill": "#007DBA",
                                  "id": "undefined-data-id",
                                },
                                "children": Array [],
                                "name": "path",
                                "namespace": "http://www.w3.org/2000/svg",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "d": undefined,
                                  "data-test-id": undefined,
                                  "fill": undefined,
                                  "id": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "d": undefined,
                                  "data-test-id": undefined,
                                  "fill": undefined,
                                  "id": undefined,
                                },
                              },
                            ],
                            "name": "svg",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "data-test-id": undefined,
                              "focusable": undefined,
                              "height": undefined,
                              "viewBox": undefined,
                              "width": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "data-test-id": undefined,
                              "focusable": undefined,
                              "height": undefined,
                              "viewBox": undefined,
                              "width": undefined,
                            },
                          },
                        ],
                        "name": "button",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-expanded": undefined,
                          "aria-haspopup": undefined,
                          "aria-label": undefined,
                          "class": undefined,
                          "data-test-id": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-expanded": undefined,
                          "aria-haspopup": undefined,
                          "aria-label": undefined,
                          "class": undefined,
                          "data-test-id": undefined,
                        },
                      },
                    ],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "data-required": "false",
                        "data-test-id": "test",
                        "data-testid": "test",
                        "id": "test",
                        "placeholder": "DD/MM/YYYY",
                        "tabindex": "0",
                        "type": "text",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "class": "sc-hMqMXs dLhBbu",
                          "data-test-id": "test-input-text-label",
                          "for": "test",
                        },
                        "children": Array [
                          Object {
                            "data": "date-picker",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "data-test-id": undefined,
                          "for": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "data-test-id": undefined,
                          "for": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "data-required": undefined,
                        "data-test-id": undefined,
                        "data-testid": undefined,
                        "id": undefined,
                        "placeholder": undefined,
                        "tabindex": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "data-required": undefined,
                        "data-test-id": undefined,
                        "data-testid": undefined,
                        "id": undefined,
                        "placeholder": undefined,
                        "tabindex": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
